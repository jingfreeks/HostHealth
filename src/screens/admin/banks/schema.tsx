import * as yup from 'yup';
import {message} from '@/config/constant'

export const Schema = yup
  .object({
    name: yup.string().required(message[100013]),
    address:yup.string().required(message[100014]),
  })
  .required();
