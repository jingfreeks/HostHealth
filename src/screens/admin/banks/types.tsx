import { RouteProp } from '@react-navigation/native';
import type {AppNavigationProps} from '@/navigation/types';
type StateFormScreenRouteProp = RouteProp<AppNavigationProps, 'BankForm'>;
export type RoutesProps = {
  route: StateFormScreenRouteProp
};

export type StateFormProps = {
  name:string;
  address:string;
  _id:string;
};
