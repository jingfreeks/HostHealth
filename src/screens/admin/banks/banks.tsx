import { View, Text } from 'react-native'
import React from 'react'

const Banks = () => {
  return (
    <View>
      <Text>Banks</Text>
    </View>
  )
}

export default Banks