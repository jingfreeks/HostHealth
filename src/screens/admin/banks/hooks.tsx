import React from 'react';
import { Alert } from 'react-native';
import {StackNavigationProp} from '@react-navigation/stack';
import type {RootNavigationProps} from '@/navigation/types';
import {useNavigation} from '@react-navigation/native';
import {useGetBanksQuery, useDeleteBanksMutation} from '@/slice';

export const useBankHooks = () => {
  const navigation = useNavigation<StackNavigationProp<RootNavigationProps>>();

  const {
    data: bank,
    isLoading,
    isSuccess,
    isError,
    error,
  } = useGetBanksQuery<{
    refetch: () => void;
    data: any;
    isLoading: boolean;
    isSuccess: string;
    isError: boolean;
    error: any;
  }>('getBanks');

  const [deleteBanks, {isLoading: deleteLoading}] = useDeleteBanksMutation();
  const handleDeleteBanks = async (id: string) => {
    try {
      await deleteBanks({
        id,
      }).unwrap();
    
    } catch (error) {
      Alert.alert(error?.data?.message);
    }
  };
  return {
    navigation,
    bank,
    isLoading,
    isSuccess,
    isError,
    error,
    handleDeleteBanks,
    deleteLoading,
  };
};
