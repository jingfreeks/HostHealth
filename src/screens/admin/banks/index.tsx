export {default as Banks} from './banks'
export {default as BankForm} from './form'