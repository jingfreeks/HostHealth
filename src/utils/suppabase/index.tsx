export {default as supabase} from './suppabase';
