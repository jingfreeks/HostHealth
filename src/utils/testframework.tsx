import React from 'react';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
// import {QueryClient, QueryClientProvider} from '@tanstack/react-query';
// // import {DARK_THEME} from '@/utils/constants';
// // import {ThemeProvider} from 'styled-components/native';

type MockProviderType = {
  children?: React.ReactNode;
  store?: object;
};
export const MockProvider = (props: MockProviderType) => {
  const {store, children} = props;
  const mockStore = configureStore([thunk]);
  const mocked = mockStore(store);
  return <Provider store={mocked}>{children}</Provider>;
};

export const testingProps = (reference: string): unknown => ({
  testID: reference,
  accessibilityLabel: reference,
});

// const queryClient = new QueryClient({
//   defaultOptions: {
//     queries: {
//       retry: false,
//     },
//   },
// });
// export const wrapper = ({children}) => (
//   <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
// );
