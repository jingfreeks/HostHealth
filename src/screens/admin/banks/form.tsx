import React from 'react';
import {
  FormHeaderContainerStyled,
  FormTextInputContainerStyled,
  FormContainerStyled,
  FormHeaderTextStyled,
} from './styles';
import {FormTextController, Bbutton} from '@/component';
import {useBankHooks} from './hooks';
import {colors} from '@/utils/themes';
import {useForm, FormProvider, SubmitHandler} from 'react-hook-form';
import {yupResolver} from '@hookform/resolvers/yup';
import {  useAddBanksMutation,
  useUpdateBanksMutation,} from '@/slice';
import {Schema} from './schema';
import * as yup from 'yup';
import type {RoutesProps} from './types';
import {testingProps} from '@/utils/testframework';

const Form = (props: RoutesProps) => {
  const {navigation} = useBankHooks();
  const {route} = props;
  const {name,address, _id} = route?.params || {};
  type FormData = yup.InferType<typeof Schema>;
  const formMethod = useForm<FormData>({
    defaultValues: {
      name: name || '',
    },
    resolver: yupResolver(Schema),
  });

  const [addBanks, {isLoading: addBanksLoading}] = useAddBanksMutation();
  const [updateBanks, {isLoading: updateBanksLoading}] = useUpdateBanksMutation();
  const onSubmit: SubmitHandler<FormData> = async data => {
    try {
      let response: any;
      if (_id) {
        //update
        response = await updateBanks({
          name: data?.name,
          address:data?.address,
          id: _id,
        }).unwrap();
      } else {
        //insert
        response = await addBanks({
          name: data?.name,
          address:data?.address,
        });
      }
      if (response?.error) {
        alert(response?.error?.data?.message);
      } else {
        navigation.goBack();
      }
    } catch (error) {
      console.log('error', error);
    }
  };
  return (
    <FormContainerStyled>
      <FormProvider {...formMethod}>
        <FormHeaderContainerStyled>
          <FormHeaderTextStyled TextMode="Htitlenormal">
            Bank Form Information
          </FormHeaderTextStyled>
        </FormHeaderContainerStyled>
        <FormTextInputContainerStyled>
          <FormTextController
            Label="Name"
            name="name"
            placeholder="Name"
            rules={{
              required: true,
            }}
          />
        </FormTextInputContainerStyled>
        <FormTextInputContainerStyled>
          <FormTextController
            Label="Address"
            name="address"
            placeholder="Address"
            rules={{
              required: true,
            }}
          />
        </FormTextInputContainerStyled>
        <Bbutton
          bcolor={colors.lightergreen}
          border={10}
          loaders={addBanksLoading || updateBanksLoading}
          title="Save"
          onPress={formMethod.handleSubmit(onSubmit)}
        />
      </FormProvider>
    </FormContainerStyled>
  );
};

export default Form;
