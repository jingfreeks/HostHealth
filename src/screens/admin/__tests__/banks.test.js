import React from 'react'
import { fireEvent } from '@testing-library/react-native';
import {renderWithProviders} from '@/utils/testframeworknew';
import {Banks,BankForm} from '../banks';


jest.mock('react-hook-form', () => ({
  ...jest.requireActual('react-hook-form'),
  useFormContext: () => ({
    handleSubmit: () => jest.fn(),
    control: {
      register: jest.fn(),
      unregister: jest.fn(),
      getFieldState: jest.fn(),
      _names: {
        array: new Set('test'),
        mount: new Set('test'),
        unMount: new Set('test'),
        watch: new Set('test'),
        focus: 'test',
        watchAll: false,
      },
      _subjects: {
        watch: jest.fn(),
        array: jest.fn(),
        state: jest.fn(),
      },
      _getWatch: jest.fn(),
      _formValues: ['test','test'],
      _defaultValues: ['name','address'],
    },
    getValues: () => {
      return [];
    },
    setValue: () => jest.fn(),
    formState: () => jest.fn(),
    watch: () => jest.fn(),
  }),
  Controller: () => [],
  useSubscribe: () => ({
    r: {current: {subject: {subscribe: () => jest.fn()}}},
  }),
}));
describe('Banks admin screen', () => {
  it('Should work as expected to get snapshot', () => {
    const all = renderWithProviders(<Banks />);
    expect(all.toJSON()).toMatchSnapshot();
  });

  it('Should to trigger Create Button', () => {
    const all = renderWithProviders(<Banks />);
    const el=all.getByTestId('BanksCreateButtonTestId')
    fireEvent(el,'onPress')
    expect(all.toJSON()).toBeTruthy();
  });
});

describe('Banks Form admin screen', () => {
  it('Should work as expected to get snapshot', () => {
    const all = renderWithProviders(<BankForm />);
    expect(all.toJSON()).toMatchSnapshot();
  });
});
