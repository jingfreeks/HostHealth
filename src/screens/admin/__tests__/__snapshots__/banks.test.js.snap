// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banks Form admin screen Should work as expected to get snapshot 1`] = `
<View
  style={
    [
      {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
        "marginLeft": 39.23529411764706,
        "marginRight": 39.23529411764706,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "marginBottom": 39.23529411764706,
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#0e3f6c",
            "fontFamily": "Avenir",
            "fontSize": 35.311764705882354,
            "fontWeight": "500",
          },
          {
            "fontSize": 30,
            "fontWeight": "900",
          },
        ]
      }
    >
      Bank Form Information
    </Text>
  </View>
  <View
    style={
      [
        {
          "marginBottom": 9.808823529411764,
          "marginTop": 39.23529411764706,
        },
      ]
    }
  />
  <View
    style={
      [
        {
          "marginBottom": 9.808823529411764,
          "marginTop": 39.23529411764706,
        },
      ]
    }
  />
  <View
    style={
      [
        {
          "marginLeft": 5,
          "marginRight": 5,
          "paddingBottom": 5,
          "paddingLeft": 5,
          "paddingRight": 5,
          "paddingTop": 5,
        },
      ]
    }
  >
    <View
      border={10}
      borderc="#99E1E1"
      borderw={2}
      style={
        [
          {
            "borderColor": "#99E1E1",
            "borderRadius": 10,
            "borderWidth": 2,
          },
        ]
      }
    >
      <View
        accessibilityLabel="bankFormButtonSubmitTestId"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#d6f3f3",
            "borderRadius": 10,
            "opacity": 1,
            "paddingBottom": 15,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 15,
          }
        }
        testID="bankFormButtonSubmitTestId"
      >
        <Text
          style={
            [
              {
                "color": "#03606A",
                "fontFamily": "Avenir",
                "fontSize": 15,
                "fontWeight": "500",
              },
            ]
          }
        >
          Save
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Banks admin screen Should work as expected to get snapshot 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <RCTSafeAreaView
    style={
      [
        {},
      ]
    }
  >
    <View
      alignType="center"
      style={
        [
          {
            "alignItems": "center",
            "alignSelf": "center",
            "flexDirection": "column",
          },
        ]
      }
    >
      <Image
        resizeMode="contain"
        size={150}
        source={
          {
            "testUri": "../../../src/assets/pcitiesempty.png",
          }
        }
        style={
          [
            {
              "height": 294.2647058823529,
              "width": 294.2647058823529,
            },
          ]
        }
      />
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "Avenir",
              "fontSize": 16,
              "fontWeight": "400",
            },
          ]
        }
      >
        No Banks matches your preferred locations
      </Text>
    </View>
  </RCTSafeAreaView>
  <View
    collapsable={false}
    style={
      {
        "alignItems": "center",
        "borderRadius": 28,
        "bottom": 0,
        "flexDirection": "row",
        "justifyContent": "center",
        "margin": 16,
        "opacity": 1,
        "position": "absolute",
        "right": 0,
        "transform": [
          {
            "scale": 1,
          },
        ],
      }
    }
  >
    <View
      style={
        [
          {
            "borderRadius": 3,
            "overflow": "hidden",
          },
          {
            "borderRadius": 3,
          },
          {
            "borderRadius": 28,
            "elevation": 4,
          },
          false,
        ]
      }
    >
      <View
        accessibilityLabel="BanksCreateButtonTestId"
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": false,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
        testID="BanksCreateButtonTestId"
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#ca71eb",
              "borderColor": "#2089dc",
              "borderRadius": 3,
              "borderWidth": 0,
              "flexDirection": "row",
              "height": 48,
              "justifyContent": "center",
              "padding": 8,
              "paddingHorizontal": 16,
            }
          }
        >
          <Text
            style={
              {
                "color": "#ffffff",
                "fontSize": 18,
                "marginHorizontal": 8,
                "paddingVertical": 1,
                "textAlign": "center",
              }
            }
          >
            Create
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
